### DISCUSSION TITLE
How to send data to the client at login and how to receive it on the client

### DISCUSSION THREAD

**Discussion by guoguo909** [FabricMC/fabric-api#1993] [Mod Dev Support] [answered]
How to send data to the client at login and how to receive it on the client.
![QQ截图20220210195759](https://user-images.githubusercontent.com/62527166/153687185-5e530283-268d-4131-9884-0c3999760df2.jpg)
Image is the code where I try to send the data.

**Comment by warjort** [comment, marked as answer]
The code you have above doesn't look correct.
````MinecraftClient```` is only available on the client so that method wouldn't work on a dedicated server.

Here's an example from the testsuite using the fabric api:
https://github.com/FabricMC/fabric/tree/1.18.2/fabric-networking-api-v1/src/testmod/java/net/fabricmc/fabric/test/networking/login
This is just a simple test where the server sends a dummy message to client which responds with a dummy message.

The full api is here:
https://github.com/FabricMC/fabric/tree/1.18.2/fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/api
with the wiki here:
https://fabricmc.net/wiki/tutorial:networking

**Reply by guoguo909** [reply]
How to write channel
